### PR TITLE
Fix --math-overflow flag and add tests

### PR DESCRIPTION
**Pull request by seanyoung**
- fix `--math-overflow` cli option
- add test for `--math-overflow`
- adds unchecked to the cfg printer

### CODE REVIEW

File: src/codegen/yul/builtin.rs [hunk 1]
```diff
@@ -229,13 +229,13 @@ fn process_arithmetic(
 
     match builtin_ty {
         YulBuiltInFunction::Add => {
-            Expression::Add(*loc, left.ty(), true, Box::new(left), Box::new(right))
+            Expression::Add(*loc, left.ty(), false, Box::new(left), Box::new(right))
```
Comment (LucasSte) [inline src/codegen/yul/builtin.rs:232]: `true` means the operation is unchecked. `false` means the operations is NOT unchecked.

Yul operation should be unchecked, so you need to leave the `true` here.

Check it here:

https://github.com/hyperledger/solang/blob/143da73c8b12b7a1ac1ed3b78280da1ae6f27273/src/emit/expression.rs#L127

Comment (LucasSte) [reply src/codegen/yul/builtin.rs:232]: Another idea is to change to nomenclature to something less confusing.

Comment (seanyoung) [reply src/codegen/yul/builtin.rs:232]: Thanks for catching that, like you say I got confused with the *un*checked. That what I get for working late!

**Review by LucasSte — APPROVED** [review]

**Comment by LucasSte** [comment]
Please, fix the failing tests before merging.

**Review by xermicus — APPROVED** [review]
